samtranslator/model/api/api_generator.py

```diff
@@ -502,6 +502,9 @@ def _construct_api_domain(self, rest_api, route53_record_set_groups):  # type: i
         else:
             basepaths = None
 
+        # Boolean to allow/disallow hyphens in BasePath property
+        strip_hyphens = self.domain.get("StripHyphens", True)
```

Comment:
For future, we should have some way of specifying the default in the props definition, so we don't need to sprinkle custom logic like this.

Reply:
Agree.